packages/ui/src/components/ValidatorList/KeystoresDataGrid.tsx

```diff
@@ -61,6 +69,106 @@ export default function KeystoresDataGrid({
     openDashboardTab();
   }, [validatorSummaryURL]);
 
+  useState<CustomValidatorUpdateRequest>();
+  const [validatorsSubscriptionMap, setValidatorsData] =
+    useState<SmoothStatusByPubkey>({});
+  const [oracleCallError, setOracleCallError] = useState<string>();
+
+  // Check that Smooth API returns an expected response format
+  function isValidOracleResponse(response: SmoothValidatorByIndexApiResponse): boolean {
+    return (
+      response &&
+      (Array.isArray(response.found_validators) ||
+        response.found_validators === null) &&
+      (Array.isArray(response.not_found_validators) ||
+        response.not_found_validators === null)
+    );
+  }
+
+  // This function fetches the subscription status of the validators in the rows array
+  // It uses the validator index to fetch the status from the Oracle API
+  // It updates the validatorsSubscriptionMap state with the new data
+  const fetchValidatorsData = async (rows: CustomValidatorGetResponse[]) => {
+    const newValidatorSubscriptionStatus: SmoothStatusByPubkey = {};
+    // Filter rows to include only those with an index
+    const rowsWithIndex = rows.filter(row => row.index);
+  
+    // Only proceed if there is at least one row with an index
+    // new validatos will not have an index yet
+    if (rowsWithIndex.length > 0) {
+      try {
+        // Initialize an array to hold the batches
+      const batches = [];
+      // Create a helper array from rowsWithIndex to avoid modifying the original array
+      const helperArray = [...rowsWithIndex];
+
+      // Split the indices into batches of 100 or less
+      while (helperArray.length) {
+        batches.push(helperArray.splice(0, 100));
+      }
+        for (const batch of batches) {
+          const apiUrl =
```

Comment:
This should be done out from the loop, it does not need to be executed several times

Reply:
true, will fix